src/oracle/types/mod.rs

```diff
@@ -101,112 +86,73 @@ macro_rules! not_none {
 
 impl HasSqlType<SmallInt> for Oracle {
     fn metadata(_: &Self::MetadataLookup) -> Self::TypeMetadata {
-        OciDataType::Number {
-            raw_type: ffi::SQLT_NUM,
-            size: 2,
-            scale: Some(0),
-            precision: 5,
-        }
+        OciDataType::SmallInt
     }
 }
 
 impl HasSqlType<Integer> for Oracle {
     fn metadata(_: &Self::MetadataLookup) -> Self::TypeMetadata {
-        OciDataType::Number {
-            raw_type: ffi::SQLT_NUM,
-            size: 4,
-            scale: Some(0),
-            precision: 10,
-        }
+        OciDataType::Integer
     }
 }
 
 impl HasSqlType<BigInt> for Oracle {
     fn metadata(_: &Self::MetadataLookup) -> Self::TypeMetadata {
-        OciDataType::Number {
-            raw_type: ffi::SQLT_NUM,
-            size: 8,
-            scale: Some(0),
-            precision: 19,
-        }
+        OciDataType::BigInt
     }
 }
 
 impl HasSqlType<Float> for Oracle {
     fn metadata(_: &Self::MetadataLookup) -> Self::TypeMetadata {
-        OciDataType::Float {
-            raw_type: ffi::SQLT_BFLOAT,
-            size: 4,
-        }
+        OciDataType::Float
     }
 }
 
 impl HasSqlType<Double> for Oracle {
     fn metadata(_: &Self::MetadataLookup) -> Self::TypeMetadata {
-        OciDataType::Float {
-            raw_type: ffi::SQLT_BDOUBLE,
-            size: 8,
-        }
+        OciDataType::Double
     }
 }
 
 impl HasSqlType<Numeric> for Oracle {
```

Comment:
Remove that imp if not supported.

(Or add a `FromSql`/`ToSql` implementation for `BigDecimal`)